src/main.rs

```diff
@@ -255,3 +264,71 @@ fn _main() -> PaziResult {
     }
     res
 }
+
+fn intercept_ctrl_c() -> PaziResult {
```

Comment:
I don't think this should return a `PaziResult`; `PaziResult` is meant to be used for exit codes rather than as a `Result`. I think `Result<(), ()>` fits this function somewhat better, with main then converting `Err(())` to a `PaziResult`.